src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -1192,4 +1226,33 @@ public String toString() {
     public LibraryTabContainer getLibraryTabContainer() {
         return tabContainer;
     }
+
+    /**
+     * Starts periodic monitoring of Git status for the database file
+     */
+    private void startGitStatusMonitoring() {
+        if (!bibDatabaseContext.isUnderVersionControl()) {
+            return;
+        }
+
+        try {
+            Runnable checkGitStatusAndUpdateUI = () -> {
+                bibDatabaseContext.getGitStatus();
+                Platform.runLater(() -> updateTabTitle(changedProperty.getValue()));
+            };
+
+            checkGitStatusAndUpdateUI.run();
+
+            PauseTransition gitStatusTimer = new PauseTransition(Duration.seconds(GIT_STATUS_REFRESH_INTERVAL_SECONDS));
+            gitStatusTimer.setOnFinished(event -> {
+                checkGitStatusAndUpdateUI.run();
+                gitStatusTimer.play();
+            });
+            gitStatusTimer.play();
+        } catch (IllegalStateException e) {
```

Comment:
This can be chained with a pipe.
Isn't there any more specific Git exception than just a very general RuntimeException?